### PR TITLE
validate with timestamp

### PR DESCRIPTION
**Pull request by vaibhavppathak**

### CODE REVIEW

File: routes/index.js [hunk 1]
```diff
@@ -67,5 +67,20 @@ router.get('/user/get/:access_token', function(req, res) {
       res.json(data);
     }
   });
+  
+<!----------- Pagination --------->
+router.get('/user/list/:page', function(req, res) {
```
Comment (mageexxcel) [inline routes/index.js]: why is code relating to paging in validate time stamp branch?

File: routes/validate.js [hunk 2]
```diff
@@ -1,18 +1,25 @@
...
-			} else if(!res1){
-				res.json('you are not authenticated');
-			}else {
-				next();
-			}
-		})
-	}else{
-		next();
-	}
+    var username = req.body.user_name;
```
Comment (mageexxcel) [inline routes/validate.js]: why is username used i dont understand?

File: routes/index.js [hunk 3]
```diff
@@ -1,71 +1,103 @@
...
+            res.json({ 'access_token': docs._id });
+        } else {
+            res.status(500).send({ status: 0, message: "Invalid password" });
+        }
     });
 });
 
-<!--------- fetch data from mongodb through url -------->
+<!---- fetch data from mongodb through url -------->
 router.get('/user/get/:access_token', function(req, res) {
```
Comment (mageexxcel) [inline routes/index.js]: this is wrong as i mentioned access_token is random. 

url is /user/get/:id

File: routes/index.js [hunk 4]
```diff
@@ -1,71 +1,103 @@
...
-  },function(err, data){
-    if (err) {
-      res.json("Invalid token");
-    }else {
-      res.json(data);
-    }
-  });
+    var access_token = req.params.access_token;
+    req.users.findOne({
+        "_id": access_token,
```
Comment (mageexxcel) [inline routes/index.js]: this will never work. access token is random

File: routes/index.js [hunk 5]
```diff
@@ -1,71 +1,103 @@
...
+        if (err) {
+            res.json("Invalid token");
+        } else {
+            res.json(data);
+        }
+    });
+});
+
+<!---- Delete data from mongodb through url  ----->
+router.get('/user/delete/:access_token', function(req, res) {
```
Comment (mageexxcel) [inline routes/index.js]: same as above, wrong

File: routes/validate.js [hunk 6]
```diff
@@ -1,18 +1,21 @@
...
-			}else {
-				next();
-			}
-		})
-	}else{
-		next();
-	}
-};
+    var access_token = req.url.split("/");
+    var startDate = moment(access_token[3], "YYYY-MM-DD'T'HH:mm:ss:SSSZ");
```
Comment (mageexxcel) [inline routes/validate.js]: how is access token related to date? access token is a random string

File: routes/validate.js [hunk 7]
```diff
@@ -1,18 +1,21 @@
...
+    var endDate = moment();
+    var Difference = endDate.diff(startDate);
+    if (Difference < 60 * 60 * 1000) {
+        req.users.findOne({
```
Comment (mageexxcel) [inline routes/validate.js]: wrong. users table doesnt have access token

File: routes/validate.js (hunk 6 above)
Comment (vaibhavppathak) [reply routes/validate.js]: here access token is login time

**Pull request merged**